src/features/advanced-content-insights/ContentInsightsSection.js

```diff
@@ -0,0 +1,145 @@
+    contentInsightsConfig,
+    item,
+    onAdvancedInsightsEmailToggle = noop,
+    onAdvancedInsightsNotificationToggle = noop,
+    onAdvancedInsightsToggle = noop,
+    renderOptionsInDropdown = false,
+    submitting,
+}: Props) {
+    // TODO: Add extension/type checking, since we only support documents
+    if (!contentInsightsConfig || !item || item.type !== ITEM_TYPE_FILE) {
+        return null;
+    }
+
+    const { isActive, requireEmail, requireNotification } = contentInsightsConfig;
+    const showDescriptionInTooltip = !renderOptionsInDropdown || isActive;
+    const helpLink = !showDescriptionInTooltip && (
+        <a href="https://support.box.com" rel="noopener noreferrer" target="_blank">
+            <FormattedMessage {...messages.learnMore} />
+        </a>
+    );
+
+    const advancedContentInsightsDescription = (
+        <FormattedMessage {...messages.contentInsightsDescription} values={{ helpLink }} />
+    );
+
+    const renderCheckboxes = () => {
+        //  Render the require email and require notification options
+        return (
+            <>
+                <CheckboxWrapper wrapComponent={renderOptionsInDropdown}>
+                    <Checkbox
+                        isChecked={requireEmail}
+                        label={<FormattedMessage {...messages.contentInsightsRequireEmail} />}
+                        data-testid="require-email"
+                        onChange={() => onAdvancedInsightsEmailToggle(!requireEmail)}
+                        onClick={event => event.stopPropagation()}
+                    />
+                </CheckboxWrapper>
+                <CheckboxWrapper wrapComponent={renderOptionsInDropdown}>
+                    <Checkbox
+                        isChecked={requireNotification}
+                        label={<FormattedMessage {...messages.contentInsightsRequireNotify} />}
+                        data-testid="require-notify"
+                        onChange={() => onAdvancedInsightsNotificationToggle(!requireNotification)}
+                        onClick={event => event.stopPropagation()}
+                    />
+                </CheckboxWrapper>
+            </>
+        );
+    };
+
+    const renderAdvancedContentInsightsOptions = () => {
+        //  Render the require email and require notification options
+        return (
+            <>
+                {!renderOptionsInDropdown && <div className="shared-link-checkgroup-row">{renderCheckboxes()}</div>}
+                {renderOptionsInDropdown && (
+                    <DropdownMenu className="ContentInsightsSection-dropdownMenu" isRightAligned>
+                        <PlainButton type="button" data-testid="insights-options-dropdown-toggle">
+                            <Gear />
+                        </PlainButton>
+                        <Menu>{renderCheckboxes()}</Menu>
+                    </DropdownMenu>
+                )}
+            </>
+        );
+    };
```

Comment:
I think it would be cleaner to have `renderCheckboxes` return an array of `Checkbox` components

then you could use a map in `<Menu>...</Menu>` like `checkboxes.map(checkbox => <MenuItem>{checkbox}</MenuItem>` which removes the need for `CheckboxWrapper`

Reply:
Dropped the Advanced Content Insights options because they will be decoupled.